matrix-sdk-android/src/main/java/org/matrix/android/sdk/api/session/room/members/MembershipService.kt

```diff
@@ -30,6 +31,20 @@ interface MembershipService {
      */
     suspend fun loadRoomMembersIfNeeded()
 
+    /**
+     * All the room members can be not loaded, for instance after an initial sync.
+     * All the members will be loaded when calling [loadRoomMembersIfNeeded], or when sending an encrypted
+     * event to the room.
+     * The fun let the app know if all the members have been loaded for this room.
+     * @return true if all the members are loaded, or false elsewhere.
+     */
+    suspend fun areAllMembersLoaded(): Boolean
+
+    /**
+     * Live version for [areAllMembersLoaded].
+     */
+    fun areAllMembersLoadedLive(): LiveData<Optional<Boolean>>
```

Comment:
Should we annotate the method with `@MainThread` as in the implementation `Monarchy` will crash if it is not called from main thread?

Reply:
I think all methods that return `LiveData` has to be called from the main thread, no?

Reply:
Indeed I guess, I commented on the wrong method, sorry. My comment was for a method exposing a `Flow` relying internally on a `LiveData` like in `FlowRoom`. `Flow` collection is normally not constraint to main thread and it can be surprising if we don't look at the implementation.